core/commands/add.go

```diff
@@ -120,7 +136,12 @@ remains to be implemented.
 					return
 				}
 
-				rootnd, err := addFile(n, file, outChan, progress, wrap, trickle)
+				// If the file is not a folder, then let's get the root of that
+				// folder and attempt to load the appropriate .ipfsignore.
+				localIgnorePatterns := checkForParentIgnorePatterns(file.FileName(), ignoreFilePatterns)
```

Comment:
the hidden logic should be fine to stay here though, we encode the files attributes and send them over. But the enumeration of which files to add is done in either `commands/cli/parse` or `commands/files`, namely the `serialFile` struct (i think) which is basically a directory and when `NextFile` is called on it, it will return the next item under it... 

This change is more difficult than I first thought it would be...